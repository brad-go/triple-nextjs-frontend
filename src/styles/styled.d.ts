import 'styled-components'

declare module 'styled-components' {
  export interface DefaultTheme {
    color: {
      white: string
      white100: string
      brightGray: string
      deepBrightGray: string
      gray: string
      gray200: string
      gray500: string
      gray700: string
      gray800: string
      mint: string
      blue: string
      gunmetal: string
    }
    boxShadow: {
      phoneFrame: string
      review: string
    }
  }
}
