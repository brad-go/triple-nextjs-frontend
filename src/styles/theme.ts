import { DefaultTheme } from 'styled-components'

export const color = {
  white: '#fff',
  white100: '#fff1A',
  brightGray: '#fafafa',
  deepBrightGray: '#f5f5f5',
  gray: '#3a3a3a',
  gray200: '#3a3a3a33',
  gray500: '#3a3a3a80',
  gray700: '#3a3a3ab3',
  gray800: '#3a3a3acc',
  mint: '#26cec2',
  blue: '#3666ff',
  gunmetal: '#233332fa',
}

export const boxShadow = {
  phoneFrame:
    'rgb(0 0 0 18%) 0px 9px 17px 0px, rgb(0 0 0 / 8%) 0px -4px 6px 0px inset',
  review: 'rgb(0 0 0 / 5%) 0px 10px 20px 0px',
}

export const deviceSizes = {
  mobile: 1141,
  pc: 1142,
}

export const device = {
  mobile: `screen and (max-width: ${deviceSizes.mobile}px)`,
  pc: `screen and (min-width: ${deviceSizes.pc}px)`,
}

export const theme: DefaultTheme = {
  color,
  boxShadow,
  device,
}
