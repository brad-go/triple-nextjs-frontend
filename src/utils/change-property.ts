export const changePropertyCamelToKebab = (property: string) => {
  const capital = property.match(/[A-Z]/g)

  if (capital) {
    const capitalIndex = property.indexOf(capital[0])
    const firstWord = property.slice(0, capitalIndex)
    const secondWord = capital[0]
      .toLocaleLowerCase()
      .concat(property.slice(capitalIndex + 1))

    property = `${firstWord}-${secondWord}`
  }

  return property
}
