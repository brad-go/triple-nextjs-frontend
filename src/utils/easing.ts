export function easeOutExpo(
  time: number,
  start: number,
  end: number,
  duration: number,
) {
  if (time === duration) {
    return start + end
  }

  return end * (-Math.pow(2, (-10 * time) / duration) + 1) + start
}
