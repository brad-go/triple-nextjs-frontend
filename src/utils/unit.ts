export const setUnit = (value: number | string, unit = 'px') => {
  if (typeof value === 'string') {
    return value
  }

  return `${value}${unit}`
}
