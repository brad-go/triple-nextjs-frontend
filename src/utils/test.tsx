import { PropsWithChildren, ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

import { theme } from 'styles/theme'

function Provider<T>({ children }: PropsWithChildren<T>) {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: Provider, ...options })

export { customRender as render }
