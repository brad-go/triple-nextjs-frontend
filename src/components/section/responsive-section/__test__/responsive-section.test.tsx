import { screen } from '@testing-library/react'

import ResponsiveSection from '../responsive-section'

import { render } from 'utils/test'
import { device } from 'styles/theme'

describe('ResponsiveSection', () => {
  it('should be rendered', () => {
    const { container } = render(<ResponsiveSection />)

    expect(container).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('should be rendered with children', () => {
    const children = <div>Example</div>
    const { container } = render(
      <ResponsiveSection>{children}</ResponsiveSection>,
    )

    expect(screen.getByText('Example')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('should be rendered with default style', () => {
    render(<ResponsiveSection data-testid="container" />)

    const container = screen.getByTestId('container')

    expect(container).toHaveStyle('position: relative')
    expect(container).toHaveStyle('margin: 0px auto')
    expect(container).toHaveStyleRule('width', '1040px', {
      media: device.pc,
    })
    expect(container).toHaveStyleRule('height', 'auto', {
      media: device.pc,
    })
    expect(container).toHaveStyleRule('width', '360px', {
      media: device.mobile,
    })
    expect(container).toHaveStyleRule('height', 'auto', {
      media: device.mobile,
    })
  })

  it('should be rendered with primary style', () => {
    render(<ResponsiveSection data-testid="container" type="primary" />)

    const container = screen.getByTestId('container')

    expect(container).toHaveStyle('position: relative')
    expect(container).toHaveStyle('margin: 0px auto')
    expect(container).toHaveStyleRule('width', '1040px', {
      media: device.pc,
    })
    expect(container).toHaveStyleRule('height', '1000px', {
      media: device.pc,
    })
    expect(container).toHaveStyleRule('width', '360px', {
      media: device.mobile,
    })
    expect(container).toHaveStyleRule('height', '700px', {
      media: device.mobile,
    })
  })

  it('should be rendered with heading style', () => {
    render(<ResponsiveSection data-testid="container" type="heading" />)

    const container = screen.getByTestId('container')

    expect(container).toHaveStyle('position: relative')
    expect(container).toHaveStyle('margin: 0px auto')
    expect(container).toHaveStyleRule('width', '1040px', {
      media: device.pc,
    })
    expect(container).toHaveStyleRule('height', '860px', {
      media: device.pc,
    })
    expect(container).toHaveStyleRule('width', '360px', {
      media: device.mobile,
    })
    expect(container).toHaveStyleRule('height', '700px', {
      media: device.mobile,
    })
  })

  it('should be rendered with footer style', () => {
    render(<ResponsiveSection data-testid="container" type="footer" />)

    const container = screen.getByTestId('container')

    expect(container).toHaveStyle('position: relative')
    expect(container).toHaveStyle('margin: 0px auto')
    expect(container).toHaveStyleRule('width', '1040px', {
      media: device.pc,
    })
    expect(container).toHaveStyleRule('height', '1000px', {
      media: device.pc,
    })
    expect(container).toHaveStyleRule('width', '360px', {
      media: device.mobile,
    })
    expect(container).toHaveStyleRule('height', '560px', {
      media: device.mobile,
    })
  })
})
