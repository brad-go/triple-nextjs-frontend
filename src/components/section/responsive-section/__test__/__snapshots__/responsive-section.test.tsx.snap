// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponsiveSection should be rendered 1`] = `
.c0 {
  position: relative;
  margin: 0px auto;
}

@media screen and (min-width:1142px) {
  .c0 {
    width: 1040px;
    height: auto;
  }
}

@media screen and (max-width:1141px) {
  .c0 {
    width: 360px;
    height: auto;
  }
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`ResponsiveSection should be rendered with children 1`] = `
.c0 {
  position: relative;
  margin: 0px auto;
}

@media screen and (min-width:1142px) {
  .c0 {
    width: 1040px;
    height: auto;
  }
}

@media screen and (max-width:1141px) {
  .c0 {
    width: 360px;
    height: auto;
  }
}

<div>
  <div
    class="c0"
  >
    <div>
      Example
    </div>
  </div>
</div>
`;
